Failed tests (11209): TestRaftAfterRemoveRange

The following test appears to have failed:

[#11209](https://circleci.com/gh/cockroachdb/cockroach/11209):

```
I0113 08:36:03.181441 1032 raft/raft.go:803  [group 1] 2 restored progress of 1 [next = 30, match = 0, state = ProgressStateProbe, waiting = false, pendingSnapshot = 0]
I0113 08:36:03.181480 1032 raft/raft.go:803  [group 1] 2 restored progress of 2 [next = 30, match = 29, state = ProgressStateProbe, waiting = false, pendingSnapshot = 0]
I0113 08:36:03.181507 1032 raft/raft.go:768  [group 1] 2 [commit: 0] restored snapshot [index: 29, term: 6]
I0113 08:36:03.239959 1032 stopper.go:264  draining; tasks left:
2      storage/replica.go:1734
--- FAIL: TestRaftAfterRemoveRange (30.26s)
panic: timed out during shutdown [recovered]
	panic: timed out during shutdown [recovered]
	panic: timed out during shutdown

goroutine 7759 [running]:
testing.tRunner.func1(0xc820078120)
	/usr/local/go/src/testing/testing.go:450 +0x171
github.com/cockroachdb/cockroach/util/leaktest.AfterTest(0x7f0fdf212e50, 0xc820078120)
	/go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:115 +0x194
github.com/cockroachdb/cockroach/storage_test.(*multiTestContext).Stop(0xc82000c3c0)
	/go/src/github.com/cockroachdb/cockroach/storage/client_test.go:288 +0x29d
github.com/cockroachdb/cockroach/storage_test.TestRaftAfterRemoveRange(0xc820078120)
	/go/src/github.com/cockroachdb/cockroach/storage/client_raft_test.go:1130 +0x866
testing.tRunner(0xc820078120, 0x1f15b40)
	/usr/local/go/src/testing/testing.go:456 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:561 +0x86d

goroutine 1 [chan receive]:
testing.RunTests(0x185b098, 0x1f14940, 0xde, 0xde, 0x1)
	/usr/local/go/src/testing/testing.go:562 +0x8ad
testing.(*M).Run(0xc820427f08, 0x1)
	/usr/local/go/src/testing/testing.go:494 +0x70
github.com/cockroachdb/cockroach/util/leaktest.TestMainWithLeakCheck(0xc820427f08)
	/go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:37 +0x21
github.com/cockroachdb/cockroach/storage_test.TestMain(0xc820427f08)
	/go/src/github.com/cockroachdb/cockroach/storage/main_test.go:36 +0x26
main.main()
	github.com/cockroachdb/cockroach/storage/_test/_testmain.go:504 +0x113

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 5 [chan receive]:
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x225a880)
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1029 +0x64
created by github.com/cockroachdb/cockroach/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:610 +0xbc

goroutine 7803 [chan receive]:
github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open.func1(0xc8205948c0)
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:123 +0x3a
created by github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:124 +0x490

goroutine 7798 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed3b0, 0xc820516a20)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7796 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed3b0, 0xc8205169e0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7790 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).processRaft.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1588 +0x8ca
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed3b0, 0xc82055f0b0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7784 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820052070, 0xc820516740)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7795 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed3b0, 0xc8205169c0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7789 [chan receive]:
github.com/cockroachdb/cockroach/util.ListenAndServe.func2()
	/go/src/github.com/cockroachdb/cockroach/util/net.go:88 +0x5b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ec700, 0xc820516900)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7804 [IO wait]:
net.runtime_pollWait(0x7f0fdf218710, 0x72, 0xc8200121a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8201ed870, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8201ed870, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc8201ed810, 0x0, 0x7f0fdf218af8, 0xc820520700)
	/usr/local/go/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc820026140, 0xc820039d20, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:254 +0x4d
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ec700, 0xc82058b650)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7743 [chan receive]:
github.com/cockroachdb/cockroach/util.ListenAndServe.func2()
	/go/src/github.com/cockroachdb/cockroach/util/net.go:88 +0x5b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ec700, 0xc8205146e0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7777 [chan receive]:
github.com/cockroachdb/cockroach/rpc.(*Server).sendResponses(0xc820501f20, 0x7f0fdf228550, 0xc820554300, 0xc82042e6c0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:300 +0xa4
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP.func1(0xc820501f20, 0x7f0fdf228550, 0xc820554300, 0xc82042e6c0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:207 +0x3f
created by github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:208 +0x89c

goroutine 7800 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed3b0, 0xc820516a60)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7741 [chan receive]:
github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open.func1(0xc820594230)
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:123 +0x3a
created by github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:124 +0x490

goroutine 7995 [semacquire]:
sync.runtime_Syncsemacquire(0xc82033a050)
	/usr/local/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc82033a040)
	/usr/local/go/src/sync/cond.go:62 +0x9b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).Quiesce(0xc820052070)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:266 +0x1f7
github.com/cockroachdb/cockroach/storage_test.(*multiTestContext).Stop.func1(0xc82000c3c0, 0xc820265a40)
	/go/src/github.com/cockroachdb/cockroach/storage/client_test.go:262 +0x2de
created by github.com/cockroachdb/cockroach/storage_test.(*multiTestContext).Stop
	/go/src/github.com/cockroachdb/cockroach/storage/client_test.go:275 +0x6f

goroutine 7813 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed6c0, 0xc820516f20)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7799 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed3b0, 0xc820516a40)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7816 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed6c0, 0xc820516f80)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7786 [select]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc8201e4600, 0xece280372, 0x3a6c66c5, 0x225a300, 0xc82058a030, 0xc820052070, 0xc8205744e0, 0xc820263e98)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:176 +0x53f
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1.1(0xc8205744e0, 0xc820384030)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:214 +0x78
github.com/cockroachdb/cockroach/storage.(*storeRangeSet).Visit(0xc820384180, 0xc8202e0900)
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:221 +0x246
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:216 +0x1e1
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820052070, 0xc820516780)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7883 [IO wait]:
net.runtime_pollWait(0x7f0fdf2181d0, 0x72, 0xc8200121a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820181410, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820181410, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8201813b0, 0xc8203cc000, 0x1000, 0x1000, 0x0, 0x7f0fe3aa2050, 0xc8200121a0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820026868, 0xc8203cc000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/rpc.(*Client).input(0xc8200aa9c0)
	/usr/local/go/src/net/rpc/client.go:109 +0xbf
created by net/rpc.NewClientWithCodec
	/usr/local/go/src/net/rpc/client.go:201 +0xd2

goroutine 7817 [select]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc820501c20, 0xece280372, 0x3a8aebb8, 0x225a300, 0xc82058a030, 0xc8201ed6c0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:176 +0x53f
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:219 +0x23f
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed6c0, 0xc820516fa0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7806 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).processRaft.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1588 +0x8ca
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed6c0, 0xc82055f750)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7844 [IO wait]:
net.runtime_pollWait(0x7f0fdf218590, 0x72, 0xc8200121a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8201804c0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8201804c0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820180460, 0xc82038b000, 0x1000, 0x1000, 0x0, 0x7f0fe3aa2050, 0xc8200121a0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8200265b0, 0xc82038b000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/rpc.(*Client).input(0xc8203b11a0)
	/usr/local/go/src/net/rpc/client.go:109 +0xbf
created by net/rpc.NewClientWithCodec
	/usr/local/go/src/net/rpc/client.go:201 +0xd2

goroutine 7791 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).Start.func3()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:566 +0x119
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed3b0, 0xc820516940)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7837 [IO wait]:
net.runtime_pollWait(0x7f0fdf2184d0, 0x72, 0xc8200121a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc82014b870, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82014b870, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc82014b810, 0xc820336000, 0x1000, 0x1000, 0x0, 0x7f0fe3aa2050, 0xc8200121a0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8200263e8, 0xc820336000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/rpc.(*Client).input(0xc8203b0780)
	/usr/local/go/src/net/rpc/client.go:109 +0xbf
created by net/rpc.NewClientWithCodec
	/usr/local/go/src/net/rpc/client.go:201 +0xd2

goroutine 7797 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed3b0, 0xc820516a00)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7783 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820052070, 0xc820516720)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7782 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820052070, 0xc820516700)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7763 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func2()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:645 +0x361
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820052070, 0xc820514760)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7807 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).Start.func3()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:566 +0x119
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed6c0, 0xc820516e80)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7762 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:626 +0x361
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820052070, 0xc820514740)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7923 [chan receive]:
github.com/cockroachdb/cockroach/storage.(*Replica).addWriteCmd(0xc820549380, 0x7f0fdf218ab8, 0xc820339aa0, 0x0, 0xba, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:961 +0x640
github.com/cockroachdb/cockroach/storage.(*Replica).handleSkippedIntents.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:1730 +0xa4d
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunAsyncTask.func1(0xc820052070, 0xc8202ba4c0, 0x17, 0xc8205bf2c0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:131 +0x58
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:132 +0x221

goroutine 7761 [select]:
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).startStats(0xc82029d0e0)
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:214 +0xca5
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).(github.com/cockroachdb/cockroach/kv.startStats)-fm()
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:201 +0x20
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ec700, 0xc82055e2e0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7811 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed6c0, 0xc820516ee0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7836 [IO wait]:
net.runtime_pollWait(0x7f0fdf218950, 0x72, 0xc8200121a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc82014bd40, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82014bd40, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc82014bce0, 0xc820458000, 0x1000, 0x1000, 0x0, 0x7f0fe3aa2050, 0xc8200121a0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8200263f8, 0xc820458000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP.func2(0xc8205014a0, 0x7f0fdf2283b0, 0xc8200263f8, 0x7f0fdf228550, 0xc82036f500, 0xc82057f640, 0xc8203b06c0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:210 +0x5d
created by github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:213 +0x90c

goroutine 7760 [select]:
github.com/cockroachdb/cockroach/storage.(*StorePool).start.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store_pool.go:213 +0x1b6
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ec700, 0xc820516220)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7812 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed6c0, 0xc820516f00)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7785 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820052070, 0xc820516760)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7948 [semacquire]:
sync.runtime_Semacquire(0xc82053ceec)
	/usr/local/go/src/runtime/sema.go:43 +0x26
sync.(*WaitGroup).Wait(0xc82053cee0)
	/usr/local/go/src/sync/waitgroup.go:126 +0xb4
github.com/cockroachdb/cockroach/storage.(*Replica).beginCmds(0xc820549380, 0xc820ad1a68, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:738 +0x507
github.com/cockroachdb/cockroach/storage.(*Replica).addWriteCmd(0xc820549380, 0x7f0fdf218ab8, 0xc8203a6c60, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:899 +0x100
github.com/cockroachdb/cockroach/storage.(*Replica).handleSkippedIntents.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:1730 +0xa4d
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunAsyncTask.func1(0xc820052070, 0xc82053e440, 0x17, 0xc820231e50)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:131 +0x58
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:132 +0x221

goroutine 7788 [IO wait]:
net.runtime_pollWait(0x7f0fdf218110, 0x72, 0xc8200121a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8201ed4f0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8201ed4f0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc8201ed490, 0x0, 0x7f0fdf218af8, 0xc820419460)
	/usr/local/go/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc820026110, 0xc820034d20, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:254 +0x4d
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ec700, 0xc82058b0b0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7890 [IO wait]:
net.runtime_pollWait(0x7f0fdf218290, 0x72, 0xc8200121a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820053cd0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820053cd0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820053c70, 0xc8202c6000, 0x1000, 0x1000, 0x0, 0x7f0fe3aa2050, 0xc8200121a0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820118730, 0xc8202c6000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP.func2(0xc820501f20, 0x7f0fdf2283b0, 0xc820118730, 0x7f0fdf228550, 0xc820554300, 0xc820234970, 0xc82042e6c0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:210 +0x5d
created by github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:213 +0x90c

goroutine 7815 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed6c0, 0xc820516f60)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7780 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820052070, 0xc8205166c0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7742 [IO wait]:
net.runtime_pollWait(0x7f0fdf218050, 0x72, 0xc8200121a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820052a70, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820052a70, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc820052a10, 0x0, 0x7f0fdf218af8, 0xc8206870a0)
	/usr/local/go/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc820118458, 0xc8202a4d20, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:254 +0x4d
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ec700, 0xc8203850b0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7808 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:626 +0x361
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed6c0, 0xc820516ea0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7744 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).processRaft.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1588 +0x8ca
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820052070, 0xc820669640)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7745 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).Start.func3()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:566 +0x119
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820052070, 0xc820514720)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7835 [chan receive]:
github.com/cockroachdb/cockroach/rpc.(*Server).sendResponses(0xc8205014a0, 0x7f0fdf228550, 0xc82036f500, 0xc8203b06c0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:300 +0xa4
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP.func1(0xc8205014a0, 0x7f0fdf228550, 0xc82036f500, 0xc8203b06c0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:207 +0x3f
created by github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:208 +0x89c

goroutine 7814 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed6c0, 0xc820516f40)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7781 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820052070, 0xc8205166e0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7843 [IO wait]:
net.runtime_pollWait(0x7f0fdf218350, 0x72, 0xc8200121a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820180530, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820180530, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8201804d0, 0xc82021b000, 0x1000, 0x1000, 0x0, 0x7f0fe3aa2050, 0xc8200121a0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8200265b8, 0xc82021b000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP.func2(0xc8201e4ba0, 0x7f0fdf2283b0, 0xc8200265b8, 0x7f0fdf228550, 0xc82036f800, 0xc82042c930, 0xc8203b10e0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:210 +0x5d
created by github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:213 +0x90c

goroutine 7819 [select]:
github.com/cockroachdb/cockroach/storage.(*idAllocator).start.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/id_alloc.go:129 +0x911
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820052070, 0xc82055fa60)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7793 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func2()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:645 +0x361
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed3b0, 0xc8205169a0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7792 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:626 +0x361
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed3b0, 0xc820516960)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7801 [select]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc8205011a0, 0xece280372, 0x3a7b6537, 0x225a300, 0xc82058a030, 0xc8201ed3b0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:176 +0x53f
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:219 +0x23f
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed3b0, 0xc820516a80)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7787 [chan receive]:
github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open.func1(0xc8202d6b90)
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:123 +0x3a
created by github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:124 +0x490

goroutine 7809 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func2()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:645 +0x361
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ed6c0, 0xc820516ec0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7842 [chan receive]:
github.com/cockroachdb/cockroach/rpc.(*Server).sendResponses(0xc8201e4ba0, 0x7f0fdf228550, 0xc82036f800, 0xc8203b10e0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:300 +0xa4
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP.func1(0xc8201e4ba0, 0x7f0fdf228550, 0xc82036f800, 0xc8203b10e0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:207 +0x3f
created by github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:208 +0x89c

goroutine 7805 [chan receive]:
github.com/cockroachdb/cockroach/util.ListenAndServe.func2()
	/go/src/github.com/cockroachdb/cockroach/util/net.go:88 +0x5b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201ec700, 0xc820516e20)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62
FAIL	github.com/cockroachdb/cockroach/storage	39.953s
=== RUN   TestBatchBasics
I0113 08:35:52.872541 1024 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.00s)
=== RUN   TestBatchGet
I0113 08:35:52.873472 1024 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.00s)
=== RUN   TestBatchMerge
I0113 08:35:52.875817 1024 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.00s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

--- a/kv/dist_sender.go
+++ b/kv/dist_sender.go
@@ -41,7 +41,6 @@ import (
 // Default constants for timeouts.
 const (
 	defaultSendNextTimeout = 10 * time.Second // for now; see #2500
-	defaultRPCTimeout      = 5 * time.Second
 	defaultClientTimeout   = 10 * time.Second
 	retryBackoff           = 250 * time.Millisecond
 	maxRetryBackoff        = 30 * time.Second
@@ -336,7 +335,7 @@ func (ds *DistSender) sendRPC(trace *tracer.Trace, rangeID roachpb.RangeID, repl
 		N:               1,
 		Ordering:        order,
 		SendNextTimeout: defaultSendNextTimeout,
-		Timeout:         defaultRPCTimeout,
+		Timeout:         rpc.DefaultRPCTimeout,
 		Trace:           trace,
 	}
 	// getArgs clones the arguments on demand for all but the first replica.

--- a/rpc/client.go
+++ b/rpc/client.go
@@ -35,6 +35,9 @@ import (
 	"github.com/cockroachdb/cockroach/util/retry"
 )
 
+// DefaultRPCTimeout is the default timeout on RPC calls.
+const DefaultRPCTimeout = 5 * time.Second
+
 const (
 	defaultHeartbeatInterval = 3 * time.Second
 

--- a/storage/client_test.go
+++ b/storage/client_test.go
@@ -276,20 +276,7 @@ func (m *multiTestContext) Stop() {
 		m.reenableTableSplits()
 	}()
 
-	select {
-	case <-done:
-	case <-time.After(30 * time.Second):
-		// If we've already failed, just attach another failure to the
-		// test, since a timeout during shutdown after a failure is
-		// probably not interesting, and will prevent the display of any
-		// pending t.Error. If we're timing out but the test was otherwise
-		// a success, panic so we see stack traces from other goroutines.
-		if m.t.Failed() {
-			m.t.Error("timed out during shutdown")
-		} else {
-			panic("timed out during shutdown")
-		}
-	}
+	<-done
 }
 
 // rpcSend implements the client.rpcSender interface. This implementation of "rpcSend" is

--- a/storage/replica.go
+++ b/storage/replica.go
@@ -1608,7 +1608,7 @@ func (r *Replica) maybeGossipFirstRange() *roachpb.Error {
 	// Gossip the cluster ID from all replicas of the first range; there
 	// is no expiration on the cluster ID.
 	if log.V(1) {
-		log.Infoc(ctx, "gossiping cluster id %s from store %d, range %d", r.store.ClusterID(),
+		log.Infoc(ctx, "gossiping cluster id %q from store %d, range %d", r.store.ClusterID(),
 			r.store.StoreID(), r.RangeID)
 	}
 	if err := r.store.Gossip().AddInfo(gossip.KeyClusterID, []byte(r.store.ClusterID()), 0*time.Second); err != nil {

--- a/storage/replica.go
+++ b/storage/replica.go
@@ -34,6 +34,7 @@ import (
 	"github.com/cockroachdb/cockroach/gossip"
 	"github.com/cockroachdb/cockroach/keys"
 	"github.com/cockroachdb/cockroach/roachpb"
+	"github.com/cockroachdb/cockroach/rpc"
 	"github.com/cockroachdb/cockroach/storage/engine"
 	"github.com/cockroachdb/cockroach/util"
 	"github.com/cockroachdb/cockroach/util/encoding"
@@ -959,8 +960,13 @@ func (r *Replica) addWriteCmd(ctx context.Context, ba roachpb.BatchRequest, wg *
 	var pErr *roachpb.Error
 	if err == nil {
 		// If the command was accepted by raft, wait for the range to apply it.
-		respWithErr := <-pendingCmd.done
-		br, pErr = respWithErr.Reply, respWithErr.Err
+		select {
+		case respWithErr := <-pendingCmd.done:
+			br, pErr = respWithErr.Reply, respWithErr.Err
+		case <-ctx.Done():
+			return br, roachpb.NewError(ctx.Err())
+		}
+
 	} else {
 		pErr = roachpb.NewError(err)
 	}
@@ -1714,8 +1720,9 @@ func (r *Replica) handleSkippedIntents(intents []intentsWithArg) {
 			if args.Method() == roachpb.EndTransaction {
 				var ba roachpb.BatchRequest
 				txn := item.intents[0].Txn
+				gcKey := r.Desc().StartKey.AsRawKey()
 				gcArgs := roachpb.GCRequest{
-					Span: roachpb.Span{Key: r.Desc().StartKey.AsRawKey()},
+					Span: roachpb.Span{Key: gcKey, EndKey: gcKey.Next()},
 				}
 				{
 					kvs, err := r.sequence.GetAllTransactionID(r.store.Engine(), txn.ID)
@@ -1732,7 +1739,9 @@ func (r *Replica) handleSkippedIntents(intents []intentsWithArg) {
 				gcArgs.Keys = append(gcArgs.Keys, roachpb.GCRequest_GCKey{Key: keys.TransactionKey(txn.Key, txn.ID)})
 
 				ba.Add(&gcArgs)
-				if _, pErr := r.addWriteCmd(ctx, ba, nil /* nil */); pErr != nil {
+				ctxWithDeadline, cancel := context.WithDeadline(ctx, time.Now().Add(rpc.DefaultRPCTimeout))
+				defer cancel()
+				if _, pErr := r.addWriteCmd(ctxWithDeadline, ba, nil /* nil */); pErr != nil {
 					log.Warningf("could not GC completed transaction: %s", pErr)
 				}
 			}

--- a/storage/store.go
+++ b/storage/store.go
@@ -902,8 +902,6 @@ func (s *Store) BootstrapRange(initialValues []roachpb.KeyValue) error {
 	return nil
 }
 
-// The following methods implement the RangeManager interface.
-
 // ClusterID accessor.
 func (s *Store) ClusterID() string { return s.Ident.ClusterID }
 

